corehq/apps/locations/bulk_management.py

```diff
@@ -315,14 +315,13 @@ def validate_and_parse_data_from_excel(self):
         actual = set(type_sheet_reader.headers)
         expected = set(LOCATION_TYPE_SHEET_HEADERS.values())
         if actual != expected:
-            message = ugettext_lazy("'types' sheet should contain exactly '{expected}' as the sheet headers. "
-                                    "'{missing}' are missing")
-            if actual - expected:
-                message = format_lazy('{}{}', message, ugettext_lazy(" '{extra}' are not recognized"))
+            missing = ", ".join(expected - actual)
+            extra = ", ".join(actual - expected)
+            message = ugettext_lazy("'types' sheet should contain headers '{expected}'. {missing}{extra}")
             raise LocationExcelSheetError(message.format(
                 expected=", ".join(expected),
-                missing=", ".join(expected - actual),
-                extra=", ".join(actual - expected),
+                missing=ugettext_lazy("'{}' are missing. ").format(missing) if missing else '',
```

Comment:
when i was suggesting to put both, i did think that this will be a challenge.
This message would look unclear in case there are no missing for example,
`'types' sheet should contain headers 'a,b,c'. are missing. b are not recognized`

may be use `none` instead. this wont be confused with python `None` and will convey the message too?

Reply:
Not following - if nothing is missing, this value will be a blank string.

Reply:
ya if its a blank string i gave an example of how it would read out. So i suggested using "none" instead of blank string so then it would say

'types' sheet should contain headers 'a,b,c'. none are missing. b are not recognized

Reply:
Right now it will say `'types' sheet should contain headers 'a,b,c'. 'd' are not recognized`, it doesn't contain "are missing".

Reply:
ah! you are right, the same names caught me only.